### DISCUSSION TITLE
Converting Smartdataframe to dict

### DISCUSSION THREAD

**Discussion by shwetabhattad-TU** [sinaptik-ai/pandas-ai#628] [Q&A]
Hi,
I am using PandasAI version - 1.0.8.
While i am converting a response which is of type SmartDataframe to dict. Its generating an error as below in Visual Studio Code.

'answer': 'Sorry, something went wrong. Try again', 'Error type': 'NotImplementedError'

However the same code is working in Google Colab.

Can someone please help to resolve this issue.
I have tried converting from SmartDataframe to Dataframe and then converting to dict. However in the process, column names are getting lost and is appearing as 0,1,2,... etc. Tried various possibilities but not working.

[dict_error.docx](https://github.com/gventuri/pandas-ai/files/12845310/dict_error.docx)

**Comment by shwetabhattad-TU** [comment]
Error i get is

C:\\Users\\sb3249832\\Desktop\\fapi\\Lib\\site-packages\\pandasai\\smart_dataframe\\abstract_df.py", line 156, in to_dict\n    raise NotImplementedError\nNotImplementedError\n'}

**Reply by gventuri** [reply]
@shwetabhattad-TU the bug has been fixed, please upgrade to a newer version! :)

**Reply by shwetabhattad-TU** [reply]
Thanks. Changing the version helped !

**Comment by mspronesti** [comment]
Do you mind providing a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)?

**Comment by shwetabhattad-TU** [comment]
question='show the last 2 rows of the data'
llm = OpenAI(api_token=apikey)
df = SmartDataframe(df,config={"llm": llm,})
answer=df.chat(question)
if (isinstance(answer, SmartDataframe)==True) :
    print(answer)
    dict_ans=answer.to_dict('record')

Converting to dict is throwing an NotImplementedError.

**Reply by mspronesti** [reply]
To make this a MRE, you also need to provide the (pandas) dataframe you are using, or a meaningful subset of rows.
